azurerm/resource_arm_monitor_diagnostic_setting.go

```diff
@@ -0,0 +1,441 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/monitor/mgmt/2018-03-01/insights"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/response"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmMonitorDiagnosticSetting() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmMonitorDiagnosticSettingCreateUpdate,
+		Read:   resourceArmMonitorDiagnosticSettingRead,
+		Update: resourceArmMonitorDiagnosticSettingCreateUpdate,
+		Delete: resourceArmMonitorDiagnosticSettingDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				// NOTE: there's no validation requirements listed for this
+				// so we're intentionally not validating this at this time.
+			},
+
+			"target_resource_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+
+			"eventhub_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
```

Comment:
I think we have a validation function for this:
```suggestion
				ValidateFunc: azure.ValidateEventHubName(),
			},
```